### PR TITLE
feat(engine): Add SMTP support to core mail action

### LINKED ISSUE #136: [Feature Request | Q&A] Use of alternative mail services
### **Is your feature request related to a problem? Please describe.**
No

### **Describe the solution you'd like**
I think 3rd party providers or SMTP solutions can be added optionally instead of Resend for mail actions. 

### **Describe alternatives you've considered**
[AWS SES](https://aws.amazon.com/tr/ses/pricing/) will be a very good service, especially in terms of price performance.

### **Additional context**
 Maybe you can create plugin logic for this part and turn it into a structure that anyone can use.

Additionally,
Companies support the existence of such plugin services. The more competition there is, the more the results are reflected on the user.

If the plugin is suitable, a draft plugin can be created and the relevant e-mail, SMS, etc. communication companies can be asked to make conversions in accordance with this draft. Or they may have already shared it on their github org. All you have to do is display it in the plugin list. (Maybe this can also be open sourced on github.)

Alternatively, it can be used in API providers such as [Sendy.co](https://sendy.co/).

### PR DESCRIPTION
**Pull request by mattdurant**
## Description

This adds support for SMTP to the core actions by implementing a core.send_email_smtp action that is not reliant on Resend for sending in on-prem or non-AWS environments.

Configuration for SMTP settings is done via environment variables, so the default docker-compose.yml file was updated to pass those to the worker container.  .env.example was also updated to include placeholders for the new variables for new installs.

## Related Tickets & Documents

Fixes #136 

## Screenshots/Recordings

No new visual content, only added to icons.tsx so that the new action has the same built-in send icon as the original send_email function

## Steps to QA

Nothing special, build and deploy, use the new action in a workflow.  I used a live SMTP server to test on my end.

### CODE REVIEW

**Comment by mattdurant** [comment]
Tiny error, need to close and re-submit.

**Comment by topher-lo** [comment]
@daryllimyt thoughts on making SMTP the core email action and moving resend to integrations?

**Comment by daryllimyt** [comment]
Thanks for taking this on @mattdurant! This is a great idea, let's get in for the 0.4.0 release

**Comment by mattdurant** [comment]
@daryllimyt @topher-lo do you want me to refactor and move the resend piece to a new integration and out of core, or did you want to handle that?

**Comment by daryllimyt** [comment]
Yes, it would be great if you could add those changes too - thanks!

**Comment by topher-lo** [comment]
@mattdurant looks like you'll have to setup `pre-commit` to pass our linting checks:
```
cd tracecat
pip install pre-commit
pre-commit install
pre-commit run --files tracecat/actions/core/email.py
```

Reference: https://pre-commit.com/

**Comment by mattdurant** [comment]
@topher-lo thanks for that, finished splitting resend out into a new namespace and it passed lint checks on my side

File: tracecat/actions/integrations/email/resend.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                    "Successfully posted email, but failed to find email in Resend after 3 retries."
+                )
+            elif email_status == "bounced":
+                raise EmailBouncedError("Email bounced")
+
+        return email_status
+
+
+@registry.register(
+    namespace="integrations.email",
```
Comment (topher-lo) [inline tracecat/actions/integrations/email/resend.py]:
```suggestion
    namespace="integrations.email.resend",
```

**Review by topher-lo — APPROVED** [review]
LGTM

**Pull request merged**